node/src/proxy_server/mod.rs

```diff
@@ -4157,6 +4732,367 @@ mod tests {
         assert!(subject.tunneled_hosts.is_empty());
     }
 
+    #[test]
+    fn handle_dns_resolve_failure_zero_hop() {
+        let system = System::new("handle_dns_resolve_failure_zero_hop");
+        let (dispatcher_mock, _, dispatcher_recording_arc) = make_recorder();
+        let (neighborhood_mock, _, neighborhood_recording_arc) = make_recorder();
+        let cryptde = main_cryptde();
+        let this_node_public_key = cryptde.public_key();
+        let mut subject = ProxyServer::new(
+            cryptde,
+            alias_cryptde(),
+            false, //meaning ZeroHop
+            Some(STANDARD_CONSUMING_WALLET_BALANCE),
+            false,
+        );
+        let stream_key = make_meaningless_stream_key();
+        let socket_addr = SocketAddr::from_str("1.2.3.4:5678").unwrap();
+        let mut dns_failure_retries_hash_map = HashMap::new();
+        let client_payload = make_request_payload(111, cryptde);
+        dns_failure_retries_hash_map.insert(
```

Comment:
If you're testing zero-hop mode, do you really want to preload with an exhausted retry count? It seems to me that zero-hop mode should never retry a DNS failure (what's the point?), so there should probably be some sort of special-casing involved when it comes to populating the `dns_failure_retries_hash_map`. Wouldn't a test like this be a perfect place to exercise that special-casing code, rather than bypassing it by preloading?

Reply:
We have tried to remove this insert.
When doing so we receive `NodeRecordMetadataMessage` but not `TransmitDataMsg`
what should the indented behaviour be when it comes to zero-hope mode and a DNS failure?